### PR TITLE
feat(auth): setup oauth system

### PR DESCRIPTION
**Pull request by mapra99**
So, this PR is setting up a basic email/password auth system. On the backend, it's using devise and doorkeeper. Devise for now is taking care of signing up the users and validating them, and we can later on add any feature such as email confirmation, tracking, etc. Doorkeeper is used to set up a OAuth provider and provides the endpoints to authenticate the API. The web app is registered as an OAuth client.

Steps to set this up:
1. Rebuild api and web
```bash
docker compose build
```

2. Run migrations
```bash
docker compose run api rails db:migrate
```

3. Run a rails console in api
```bash
docker compose run api rails console
```

4. Create an OAuth app
```bash
app = Doorkeeper::Application.create(name:` "MyApp", redirect_uri: "urn:ietf:wg:oauth:2.0:oob", scopes: ["read", "write"])
app.secret
app.client_id
```

5. Exit the console. Copy the app secret and client IDs and set them in a .env file on web
```bash
echo "FEEDBACKR_API_BASE_URL=http://api:3000" >> web/.env.local
echo "OAUTH_CLIENT_ID=<client id>" >> web/.env.local
echo "OAUTH_CLIENT_SECRET=<client secret>" >> web/.env.local
```

6. Run services
```bash
docker compose up
```

7. Go to http://localhost:3000 and sign up

It's funny how much I underestimated this, and I ended up creating a very complicated solution and spent a lot of time. I'll leave this gif for the future to remember that
rick-morty

### CODE REVIEW

File: web/src/app/loading.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import Spinner from "@/components/spinner"
+
+export default function Loading() {
+  return(
+    <div className="flex flex-col items-center justify-center min-h-screen">
+      <Spinner />
+    </div>
+  )
+}
```
Comment (mapra99) [inline web/src/app/loading.tsx]:
```suggestion
}

```

File: web/src/components/form/form.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      { children }
+
+      { loading ? (
+        <div className="absolute top-0 left-0 bottom-0 right-0 flex items-center justify-center bg-white/70">
+          <Spinner />
+        </div>
+      ) : null }
+    </form>
+  )
+}
```
Comment (mapra99) [inline web/src/components/form/form.tsx]:
```suggestion
}

```

File: web/src/components/signup-form/index.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+export { default } from './signup-form'
```
Comment (mapra99) [inline web/src/components/signup-form/index.ts]:
```suggestion
export { default } from './signup-form'

```

**Review by ImMPrada — APPROVED** [review]
go go go! 

BUT JUST A COUPLE OF RUBOCOP MESSAGES! USELESS MESSAGES, I THINK

**Pull request merged**